app/controllers/products_controller.rb

```diff
@@ -0,0 +1,66 @@
+class ProductsController < ApplicationController
+
+before_action only: [:show, :edit, :update, :retire] { @product = Product.find(params[:id]) }
+
+  def index
+    if params[:search]
+      @products = Product.search(params[:search]).where(retire: false).order(stock: :desc)
```

Comment:
Seems like both of these search/wheres should be in model methods. Maybe something like `search_active(search_term)` and `get_active` which will both add the `retire: false` where param and the ordering